### ISSUE TITLE
Consider forms of configuration for app specific behavior

### ISSUE THREAD

**Issue by benjaminapetersen** [patternfly/patternfly-3#727]
Original discussion on [this PR](https://github.com/patternfly/patternfly/pull/717#discussion_r135292347), regarding features implemented to support specific apps.  

`src/js/patternfly-functions-vertical-nav.js` is using local storage in a specific way to save state across browser restarts.  This behavior would likely be strange or unexpected in a different app without the same needs.  

Two items worth addressing:
- [ ] provide the ability to set `localStorage` or `sessionStorage` for this component so that consumers have the option to opt in to long term persistence.
- [ ] Consider adding a block to the `Contributing` section for features that may be app specific, but ensure they are sufficiently configurable so that other consumers can opt in/opt out of unique features.

**Comment by andresgalante** [comment]
@jeff-phillips-18 Can you please review this one when you have time please?

**Comment by jeff-phillips-18** [comment]
Yeah, I think this is worth pursuing.

**Comment by dabeng** [comment]
@dgutride  @jeff-phillips-18 I tried to append a [new option](https://github.com/dabeng/patternfly/commit/50e8a83238aa220e01d259182217e58013e816bf#diff-3a08537993f8159a012ac07ee723e26cR11) for method setupVerticalNavigation(). Here's the [demo page](https://rawgit.com/dabeng/patternfly/ls-option-dist/dist/tests/vertical-navigation-with-tertiary-pins-disable-localstorage.html). In this page, developers can use the following sentence to disable localStorage cache:
```js
$().setupVerticalNavigation(true, undefined, { rememberOpenState: false });
```

**Comment by dgutride** [comment]
@dabeng  - I think the request was to make this configurable to be able to use a different form of storage (in addition to disabling storage altogether) - can you take another stab at it?

**Comment by dabeng** [comment]
Thanks a lot @dgutride I have updated the [demo](https://rawgit.com/dabeng/patternfly/ls-option-dist/dist/tests/) page according to your advice. You can access the following links to check modifications:
<img width="559" alt="screen shot 2018-07-23 at 4 20 50 pm" src="https://user-images.githubusercontent.com/5689075/43065516-acebc8be-8e94-11e8-8166-8100dc9af694.png">

**Comment by dgutride** [comment]
@dabeng  - this looks like the right direction - one change, please use full words for the option instead of L or S (so localStorage or sessionStorage) and then put up a PR.

**Comment by dabeng** [comment]
@dgutride Thanks a lot for you advice. I have put up a [PR](https://github.com/patternfly/patternfly/pull/1109) according to your comments :blush:

**Comment by LHinson** [comment]
Closing since this has been addressed with #1109
